### PR TITLE
feat: add simple "expectEmit"

### PR DESCRIPTION
**Pull request by PaulRBerg**
Implements https://github.com/foundry-rs/forge-std/issues/252.

I know that this has been previously discussed in https://github.com/foundry-rs/forge-std/pull/42, but I think that this abstraction would be so helpful that it's worth reconsidering it.

FWIW: I have this function defined in like ~7 separate repos.

### CODE REVIEW

**Comment by mds1** [comment]
Wdyt of PR'ing this as a native cheat? It should be very simple to implement, and per https://github.com/foundry-rs/foundry/issues/3782 it'd be ideal to start moving things there by default when we can

**Comment by PaulRBerg** [comment]
> Wdyt of PR'ing this as a native cheat? 

Done!

https://github.com/foundry-rs/foundry/pull/4479

> and per https://github.com/foundry-rs/foundry/issues/3782 it'd be ideal

I'm in favor of that proposal!

**Comment by PaulRBerg** [comment]
Since I ended up implementing this as a native cheat in Foundry (https://github.com/foundry-rs/foundry/pull/4479), I refactored this branch to add the `expectEmit` cheat to `Vm.sol`, and then I force pushed to this branch.

File: src/Vm.sol [hunk 1]
```diff
@@ -326,12 +326,21 @@ interface Vm is VmSafe {
...
     // Prepare an expected log with (bool checkTopic1, bool checkTopic2, bool checkTopic3, bool checkData).
     // Call this function, then emit an event, then call a function. Internally after the call, we check if
-    // logs were emitted in the expected order with the expected topics and data (as specified by the booleans)
-    function expectEmit(bool checkTopic1, bool checkTopic2, bool checkTopic3, bool checkData) external;
-    function expectEmit(bool checkTopic1, bool checkTopic2, bool checkTopic3, bool checkData, address emitter)
-        external;
+    // logs were emitted in the expected order with the expected topics and data (as specified by the booleans).
+    // Second form also checks supplied address against emitting contract.
+    function expectEmit(bool, bool, bool, bool) external;
+    function expectEmit(bool, bool, bool, bool, address) external;
```
Comment (mds1) [inline src/Vm.sol:342]: Thanks! Will wait until the upstream PR is merged before merging this. Looks like the named parameters got dropped though?

Comment (PaulRBerg) [reply src/Vm.sol:342]: Whoops! Will add back shortly.

**Pull request merged**